plugins/inputs/openstack/openstack.go

```diff
@@ -0,0 +1,1059 @@
+// Package openstack implements an OpenStack input plugin for Telegraf
+//
+// The OpenStack input plug is a simple two phase metric collector.  In the first
+// pass a set of gatherers are run against the API to cache collections of resources.
+// In the second phase the gathered resources are combined and emitted as metrics.
+//
+// No aggregation is performed by the input plugin, instead queries to InfluxDB should
+// be used to gather global totals of things such as tag frequency.
+package openstack
+
+import (
+	"crypto/sha256"
+	"crypto/tls"
+	"encoding/hex"
+	"fmt"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/extensions/schedulerstats"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/extensions/volumetenants"
+	"github.com/gophercloud/gophercloud/openstack/blockstorage/v2/volumes"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/aggregates"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/diagnostics"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/hypervisors"
+	nova_services "github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/services"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/projects"
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/services"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/agents"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/networks"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/ports"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	"github.com/gophercloud/gophercloud/openstack/orchestration/v1/stacks"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
```

Comment:
It's a matter of taste, but other plugins group the import into "builtin" packages, external github packages and telegraf internal packages. Not a must, but easier if you need to read multiple plugins...